aws/resource_aws_batch_job_definition_test.go

```diff
@@ -0,0 +1,264 @@
+					testAccCheckJobDefinitionRecreated(t, &before, &after),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckBatchJobDefinitionExists(n string, jd *batch.JobDefinition) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		log.Printf("State: %#v", s.RootModule().Resources)
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Batch Job Queue ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).batchconn
+		arn := rs.Primary.Attributes["arn"]
+		def, err := getJobDefinition(conn, arn)
+		if err != nil {
+			return err
+		}
+		if def == nil {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		*jd = *def
+
+		return nil
+	}
+}
+
+func testAccCheckBatchJobDefinitionAttributes(jd *batch.JobDefinition, compare *batch.JobDefinition) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if !strings.HasPrefix(*jd.JobDefinitionName, "tf_acctest_batch_job_definition") {
+			return fmt.Errorf("Bad Job Definition name: %s", *jd.JobDefinitionName)
+		}
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "aws_batch_job_definition" {
+				continue
+			}
+			if *jd.JobDefinitionArn != rs.Primary.Attributes["arn"] {
+				return fmt.Errorf("Bad Job Definition ARN\n\t expected: %s\n\tgot: %s\n", rs.Primary.Attributes["arn"], *jd.JobDefinitionArn)
+			}
+			if compare != nil {
+				if compare.Parameters != nil && !reflect.DeepEqual(compare.Parameters, jd.Parameters) {
+					return fmt.Errorf("Bad Job Definition Params\n\t expected: %v\n\tgot: %v\n", compare.Parameters, jd.Parameters)
+				}
+				if compare.RetryStrategy != nil && *compare.RetryStrategy.Attempts != *jd.RetryStrategy.Attempts {
+					return fmt.Errorf("Bad Job Definition Retry Strategy\n\t expected: %d\n\tgot: %d\n", *compare.RetryStrategy.Attempts, *jd.RetryStrategy.Attempts)
+				}
+				if compare.ContainerProperties != nil && compare.ContainerProperties.Command != nil && !reflect.DeepEqual(compare.ContainerProperties, jd.ContainerProperties) {
+					return fmt.Errorf("Bad Job Definition Container Properties\n\t expected: %s\n\tgot: %s\n", compare.ContainerProperties, jd.ContainerProperties)
+				}
+			}
+		}
+		return nil
+	}
+}
+
+func testAccCheckJobDefinitionRecreated(t *testing.T,
+	before, after *batch.JobDefinition) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if *before.Revision == *after.Revision {
+			t.Fatalf("Expected change of JobDefinition Revisions, but both were %v", before.Revision)
+		}
+		return nil
+	}
+}
+
+func testAccCheckBatchJobDefinitionDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_batch_job_definition" {
+			continue
+		}
+		conn := testAccProvider.Meta().(*AWSClient).batchconn
+		js, err := getJobDefinition(conn, rs.Primary.Attributes["arn"])
+		if err == nil {
+			if *js.Status != "INACTIVE" {
```

Comment:
Really a nitpick, but positive conditionals tend to be easier to read, i.e. `*js.Status == "ACTIVE"`.